arch/xtensa/core/cpu_idle.c

```diff
@@ -6,48 +6,14 @@
 #include <zephyr/toolchain.h>
 #include <zephyr/tracing/tracing.h>
 
-/* xt-clang removes any NOPs more than 8. So we need to set
- * no optimization to avoid those NOPs from being removed.
- *
- * This function is simply enough and full of hand written
- * assembly that optimization is not really meaningful
- * anyway. So we can skip optimization unconditionally.
- * Re-evalulate its use and add #ifdef if this assumption
- * is no longer valid.
- */
+#ifndef CONFIG_ARCH_CPU_IDLE_CUSTOM
 __no_optimization
 void arch_cpu_idle(void)
 {
 	sys_trace_idle();
-
-	/* Just spin forever with interrupts unmasked, for platforms
-	 * where WAITI can't be used or where its behavior is
-	 * complicated (Intel DSPs will power gate on idle entry under
-	 * some circumstances)
-	 */
-	if (IS_ENABLED(CONFIG_XTENSA_CPU_IDLE_SPIN)) {
-		__asm__ volatile("rsil a0, 0");
-		__asm__ volatile("loop_forever: j loop_forever");
-		return;
-	}
-
-	/* Cribbed from SOF: workaround for a bug in some versions of
-	 * the LX6 IP.  Preprocessor ugliness avoids the need to
-	 * figure out how to get the compiler to unroll a loop.
-	 */
-	if (IS_ENABLED(CONFIG_XTENSA_WAITI_BUG)) {
-#define NOP4 __asm__ volatile("nop; nop; nop; nop");
-#define NOP32 NOP4 NOP4 NOP4 NOP4 NOP4 NOP4 NOP4 NOP4
-#define NOP128() NOP32 NOP32 NOP32 NOP32
-		NOP128();
-#undef NOP128
-#undef NOP32
-#undef NOP4
-		__asm__ volatile("isync; extw");
-	}
```

Comment:
Please also remove the corresponding kconfigs `CONFIG_XTENSA_CPU_IDLE_SPIN` and `CONFIG_XTENSA_WAITI_BUG`, as the intended behavior will now be done via the custom idle function.

Reply:
CAVS platform includes 1.8, 2.1, 2.5, some platforms need this workaround others don't, so I don't remove it.

Reply:
These kconfigs are no longer being used at the arch layer, so they need to be removed from there.

And since this is moved inside CAVS SoC layer, you can now use #ifdef based on cavs version to enable the behavior.

Reply:
@dcpleung  there are used by arch/xtensa/core/xtensa-asm2.c, so I can't remove them

Reply:
I just grepped and these two Kconfig were not being used in that file.

Reply:
thanks, updated